drivers/ethernet/dsa_nxp_imx_netc.c

```diff
@@ -0,0 +1,297 @@
+
+struct dsa_netc_data {
+	int port_num;
+	int port_init_count;
+	int iface_init_count;
+	swt_config_t swt_config;
+	swt_handle_t swt_handle;
+	const struct device *dev_master;
+	netc_cmd_bd_t *cmd_bd;
+};
+
+struct dsa_netc_slave_config {
+	/** MAC address for each LAN{123.,} ports */
+	uint8_t mac_addr[6];
+	const struct device *phy_dev;
+	netc_hw_mii_mode_t phy_mode;
+	volatile bool pseudo_mac;
+	const struct pinctrl_dev_config *pincfg;
+	int port_idx;
+};
+
+int dsa_netc_port_init(const struct device *dev)
+{
+	const struct dsa_netc_slave_config *cfg = dev->config;
+	struct dsa_context *ctx = dev->data;
+	struct dsa_netc_data *prv = PRV_DATA(ctx);
+	swt_config_t *swt_config = &prv->swt_config;
+
+	status_t result;
+	int err;
+
+	/* Get default config for whole switch before first port init */
+	if (prv->port_init_count == 0) {
+		SWT_GetDefaultConfig(swt_config);
+		swt_config->bridgeCfg.dVFCfg.portMembership = 0x0;
+	}
+
+	prv->port_init_count++;
+
+	if (!cfg->pseudo_mac) {
+		err = pinctrl_apply_state(cfg->pincfg, PINCTRL_STATE_DEFAULT);
+		if (err) {
+			return err;
+		}
+	}
+
+	/* miiSpeed and miiDuplex will get set correctly when link is up */
+	swt_config->ports[cfg->port_idx].ethMac.miiMode = cfg->phy_mode;
+
+	swt_config->bridgeCfg.dVFCfg.portMembership |= (1 << cfg->port_idx);
+
+	swt_config->ports[cfg->port_idx].bridgeCfg.enMacStationMove = true;
+
+	/* Switch init after all ports init */
+	if (prv->port_init_count == prv->port_num) {
+		swt_config->cmdRingUse = 1U;
+		swt_config->cmdBdrCfg[0].bdBase = prv->cmd_bd;
+		swt_config->cmdBdrCfg[0].bdLength = 8U;
+
+		result = SWT_Init(&prv->swt_handle, &prv->swt_config);
+		if (result != kStatus_Success) {
+			return -EIO;
+		}
+	}
+
+	return 0;
+}
+
+static int dsa_netc_sw_write_reg(const struct device *dev, uint16_t reg_addr, uint8_t value)
+{
+	/* This API is for PHY switch. Not available here. */
+	return 0;
+}
+
+static int dsa_netc_sw_read_reg(const struct device *dev, uint16_t reg_addr, uint8_t *value)
+{
+	/* This API is for PHY switch. Not available here. */
+	return 0;
+}
+
+/**
+ * @brief Set entry to DSA MAC address table
+ *
+ * @param dev DSA device
+ * @param mac The MAC address to be set in the table
+ * @param fw_port Port number to forward packets
+ * @param tbl_entry_idx The index of entry in the table
+ * @param flags Flags to be set in the entry
+ *
+ * @return 0 if ok, < 0 if error
+ */
+static int dsa_netc_set_mac_table_entry(const struct device *dev, const uint8_t *mac,
+					uint8_t fw_port, uint16_t tbl_entry_idx, uint16_t flags)
+{
+	return 0;
+}
+
+/**
+ * @brief Get DSA MAC address table entry
+ *
+ * @param dev DSA device
+ * @param buf The buffer for data read from the table
+ * @param tbl_entry_idx The index of entry in the table
+ *
+ * @return 0 if ok, < 0 if error
+ */
+static int dsa_netc_get_mac_table_entry(const struct device *dev, uint8_t *buf,
+					uint16_t tbl_entry_idx)
+{
+	return 0;
+}
```

Comment:
why are there docstrings here if these are just static implementation of something and returning 0 (which BTW maybe should return -ENOTSUP

Reply:
Will keep only functions and return -ENOTSUP.
This is todo work.

Thanks.